reference/forms/types/collection.rst

```diff
@@ -286,6 +286,22 @@ the empty values will be kept.
     Read about the :ref:`form's empty_data option <reference-form-option-empty-data>`
     to learn why this is necessary.
 
+A value is deleted from the collection only if the normalized value is ``null``.
+However, you can also set the option value to a ``callable``, which will be called
+for each value in the submitted collection. The ``callable`` should return
+whether or not a value must be removed from the collection. For example::
+
+    use Symfony\Component\Form\Extension\Core\Type\CollectionType;
+    // ...
+
+    $builder->add('users', CollectionType::class, array(
+        // ...
+        'delete_empty' => function ($obj = null) {
+            return null === $obj || empty($obj->firstName);
+        },
```

Comment:
The `users` field suggests a collection of type `User`, right? What do you think if we use typehint, `$user` argument and `getFirstName()` method?
```
'delete_empty' => function (User $user = null) {
    return null === $user || empty($user->getFirstName());
},
```
Something like this would be good for you?

Reply:
Yes! I've updated the code example.